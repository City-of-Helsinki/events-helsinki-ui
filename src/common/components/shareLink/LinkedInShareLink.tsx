import React from "react";
import { useTranslation } from "react-i18next";

import { ReactComponent as LinkedInIcon } from "../../../assets/icons/svg/linkedin.svg";
import ShareLinkBase from "./ShareLinkBase";
import { ShareLinkProps } from "./types";

const linkedInShareUrl = "https://linkedin.com/shareArticle";

const LinkedInShareLink: React.FC<ShareLinkProps> = ({ sharedLink }) => {
  const { t } = useTranslation();
  const queryParameters = { url: sharedLink };
  const linkLabel = t("commons.shareLink.shareOnLinkedIn");

  return (
    <ShareLinkBase
      url={linkedInShareUrl}
      queryParameters={queryParameters}
      windowName={linkLabel}
      linkLabel={linkLabel}
      icon={<LinkedInIcon />}
    />
  );
};

export default LinkedInShareLink;
