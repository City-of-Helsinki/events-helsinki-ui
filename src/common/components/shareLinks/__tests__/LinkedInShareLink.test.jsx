import { render } from "@testing-library/react";
import React from "react";

import LinkedInShareLink from "../LinkedInShareLink";

const getWrapper = props => render(<LinkedInShareLink {...props} />);

test("should apply aria label", () => {
  const sharedLink = "https://helsinki.fi/some/";
  const { getByLabelText } = getWrapper({ sharedLink });

  expect(getByLabelText("Jaa LinkedIniss√§"));
});

test("<LinkedInShareLink /> matches snapshot", () => {
  const sharedLink = "https://helsinki.fi/some/";
  const { container } = getWrapper({ sharedLink });

  expect(container.firstChild).toMatchSnapshot();
});
