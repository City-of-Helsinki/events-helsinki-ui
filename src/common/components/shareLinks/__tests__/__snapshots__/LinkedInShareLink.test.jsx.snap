// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkedInShareLink /> matches snapshot 1`] = `
<button
  aria-label="Jaa LinkedInissÃ¤"
  type="button"
>
  <svg>
    linkedin.svg
  </svg>
</button>
`;
